pandas/core/groupby/groupby.py

```diff
@@ -2431,8 +2431,11 @@ def _get_cythonized_result(
         grouper = self.grouper
 
         labels, _, ngroups = grouper.group_info
-        output: Dict[base.OutputKey, np.ndarray] = {}
+        output: Dict[Union[base.OutputKey, str], Union[np.ndarray, str]] = {}
```

Comment:
I don't like this at all, output is very well defined here and we are now using it for multiple different things.

can we just pass the index name to _wrap_aggregated_output and _wrap_transformed_output ?

Reply:
I'll try to implement this today or tomorrow. 

We have to change the Series code too in this case, when I remember correctly. But I can tell you more after implementing this. Thanks for the feedback,